server/src/internalClusterTest/java/org/elasticsearch/index/mapper/DynamicMappingIT.java

```diff
@@ -213,6 +214,69 @@ public void onFailure(Exception e) {
         }
     }
 
+    public void testTotalFieldsLimitWithRunTimefields() {
+        Settings indexSettings = Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey(), 4)
+            .build();
+
+        String mapping = """
+                {
+                  "dynamic":"runtime",
+                  "runtime": {
+                    "my_object.rfield1": {
+                       "type": "keyword"
+                    },
+                    "rfield2": {
+                      "type": "keyword"
+                    }
+                  },
+                  "properties": {
+                    "field3" : {
+                      "type": "keyword"
+                    }
+                  }
+                }
+            """;
+
+        client().admin().indices().prepareCreate("index1").setSettings(indexSettings).setMapping(mapping).get();
+        ensureGreen("index1");
+
+        {
+            // introduction of a new object with 2 sub-fields fails
+            final IndexRequestBuilder indexRequestBuilder = client().prepareIndex("index1")
+                .setId("1")
+                .setSource("my_object2", Map.of("field1", "value1", "field2", "value2"));
```

Comment:
the doc here holds only new fields? Should we add some already known fields and some new fields?

Reply:
Great suggestion, addressed in 1ee42705cb6ad4c4c5b6083a33c042a1b8269912